### PR TITLE
fix: update bin scripts with timezone support

### PR DESCRIPTION
**Pull request by rpcross**

### CODE REVIEW

File: backend/mlarchive/bin/compare.py
Comment (rjsparks) [inline backend/mlarchive/bin/compare.py:30]: Be ready to change all of the notion of when we do things to be UTC

File: backend/mlarchive/bin/scan_all.py
Comment (rjsparks) [inline backend/mlarchive/bin/scan_all.py:1015]: maybe not the best name for this function

**Review by rjsparks — APPROVED** [review]
A couple of comments on things spotted while I was here.
Please wait for Jennifer's review before merging

**Review by jennifer-richards — CHANGES_REQUESTED** [review]
A couple comments inline. I think you can probably do away with the pacific timezone reference that @rjsparks commented on because I think the underlying timezone handling is independent of that.

File: backend/mlarchive/bin/compare.py
Comment (rpcross) [reply backend/mlarchive/bin/compare.py:30]: Using Pacific timezone here to match the timezone of the server OS / filesystem. Are you saying this will change to UTC? With new infrastructure?

File: backend/mlarchive/bin/scan_all.py
Comment (rpcross) [reply backend/mlarchive/bin/scan_all.py]: Addressed in commit b62adb5c2d90f16ac4713b0c6dfc3a86b0a1c0b2

File: backend/mlarchive/bin/compare.py
Comment (rpcross) [reply backend/mlarchive/bin/compare.py:70]: In my experience this method is robust. parse returns a naive datetime when not provided with a timezone, which is assigned the default, UTC. When provided with a timezone, it is respected and converted to UTC.
I am not sure why this differs from above.
```
>>> parse("2023-09-22T11:31")
datetime.datetime(2023, 9, 22, 11, 31)
>>> str(parse("2023-09-22T11:31").astimezone(datetime.timezone.utc))
'2023-09-22 11:31:00+00:00'
>>> str(parse("2023-09-22T11:31Z").astimezone(datetime.timezone.utc))
'2023-09-22 11:31:00+00:00'
```
Which works well when supported timezone syntax is used.
```
>>> str(parse("2023-09-22T11:31-07:00").astimezone(datetime.timezone.utc))
'2023-09-22 18:31:00+00:00'
```
I can switch to use fromisoformat if that helps.

File: backend/mlarchive/bin/compare.py
Comment (rpcross) [reply backend/mlarchive/bin/compare.py]: Addressed in commit e7077ab17e8b0744cf97c46caf25338913c23f22

File: backend/mlarchive/bin/compare.py
Comment (jennifer-richards) [reply backend/mlarchive/bin/compare.py:70]: I believe the difference is because the default that `parse()` uses is the local timezone of the machine it's running on. On my system, it's ADT and I get
```python
>>> str(parse("2023-09-22T11:31").astimezone(datetime.timezone.utc))  # system time is ADT
'2023-09-22 14:31:00+00:00'
```
on ietfa, it's PDT and I get
```python
>>> str(parse("2023-09-22T11:31").astimezone(datetime.timezone.utc))  # system time is PDT
'2023-09-22 18:31:00+00:00'
```
and on your system, time is apparently UTC and the result matches. I don't think there's any way to tell it what to use as the default, it just uses system time.

If you want to guarantee that it's either UTC or whatever the user specified, you probably have to do an explicit check for whether you got a naive datetime and `replace()` only if it doesn't already have a tzinfo. Most succinct I can think of is
```python
dt = parse(...)
dt = dt.replace(tzinfo=datetime.timezone.utc if dt.tzinfo is None else dt.tzinfo)
```

`fromisoformat()` has the same behavior with respect to system time, it just saves you a dependency if you only mean to allow the iso format and don't want the additional flexibility that `dateutil` affords.

Comment (jennifer-richards) [reply backend/mlarchive/bin/compare.py:70]: (docker containers typically run in UTC regardless of what system they're on, I believe)

**Review by jennifer-richards — APPROVED** [review]
I think my complaint is resolved enough - my worry boils down to the help strings stating "UTC", which might be misleading if this ever runs with a different system time zone.

**Pull request merged**
